### PR TITLE
New SVC and SVR Patching; Move patching tests to sklearnex

### PR DESCRIPTION
**Pull request by owerbat**

### CODE REVIEW

File: sklearnex/svm/_classes.py [hunk 1]
```diff
@@ -0,0 +1,255 @@
+#===============================================================================
```
Comment (PetrovKP) [inline sklearnex/svm/_classes.py:1]: I think need separate on svc.py and svr.py files. Or rename to svm.py

File: sklearnex/svm/_classes.py [hunk 2]
```diff
@@ -0,0 +1,255 @@
...
+                          gamma=self.gamma, coef0=self.coef0,
+                          tol=self.tol, shrinking=self.shrinking,
+                          cache_size=self.cache_size,
+                          max_iter=self.max_iter)
+
+    def fit(self, X, y, sample_weight=None):
+        if self.kernel in ['linear', 'rbf', 'poly'] and not sp.isspmatrix(X):
+            logging.info("sklearn.svm.SVR.fit: " + get_patch_message("onedal"))
+
+            self._onedal_model = self._create_svr_model()
```
Comment (PetrovKP) [inline sklearnex/svm/_classes.py:83]:
```suggestion
            self._onedal_model = self._create_onedal_svr()
```

File: sklearnex/dispatcher.py [hunk 3]
```diff
@@ -16,19 +16,46 @@
...
+import sklearn.svm as svm_module
+
+# Other imports
+from functools import lru_cache
+from daal4py.sklearn._utils import daal_check_version
+
+
+@lru_cache(maxsize=None)
+def _get_map_of_algorithms_sklearnex():
+    mapping = {'svc': [[(svm_module, 'SVC', SVC_sklearnex), None]],}
```
Comment (PetrovKP) [inline sklearnex/dispatcher.py:33]: SVC_sklearnex not exist in 2021.2

File: sklearnex/dispatcher.py [hunk 4]
```diff
@@ -16,19 +16,46 @@
...
+
+    if daal_check_version((2021, 'P', 200)):
```
Comment (PetrovKP) [inline sklearnex/dispatcher.py:35]:
```suggestion
    if daal_check_version((2021, 'P', 300)):

```

File: sklearnex/svm/_classes.py (hunk 1 above)
Comment (owerbat) [reply sklearnex/svm/_classes.py:1]: Don't we want to be consistent with scikit-learn in files' names?

**Comment by owerbat** [comment]
/intelci: run
set infra_branch=dev/ntimakin-new_patching

File: sklearnex/svm/_classes.py (hunk 1 above)
Comment (PetrovKP) [reply sklearnex/svm/_classes.py:1]: Well, they have a classes_ with the context of svm.  why be 1 in 1 in such places?

**Comment by owerbat** [comment]
/intelci: run
set infra_branch=dev/ntimakin-new_patching

**Comment by owerbat** [comment]
CI: http://intel-ci.intel.com/eba36a5e-bbd1-f1f0-88f0-8cdcd4b723a1

File: deselected_tests.yaml [hunk 5]
```diff
@@ -26,13 +26,47 @@
 
 deselected_tests:
 
-  # Deselecting 5 SVC related tests where duplicate samples end up being support vectors
-  # See: https://github.com/scikit-learn/scikit-learn/issues/12738
-  - svm/tests/test_svm.py::test_svc_clone_with_callable_kernel >=0.23
-  - svm/tests/test_svm.py::test_precomputed
-  - svm/tests/test_sparse.py::test_svc >=0.23
-  - svm/tests/test_sparse.py::test_sparse_realdata
+  # Model with sparce data are different between oneAPI Data Analytics Library (oneDAL) and stock sklearn
```
Comment (OnlyDeniko) [inline deselected_tests.yaml:29]:
```suggestion
  # Model with sparse data are different between oneAPI Data Analytics Library (oneDAL) and stock sklearn
```

File: deselected_tests.yaml [hunk 6]
```diff
@@ -26,13 +26,47 @@
...
+  - svm/tests/test_sparse.py::test_svc
+  - svm/tests/test_sparse.py::test_unsorted_indices
   - svm/tests/test_sparse.py::test_svc_iris
+  - svm/tests/test_sparse.py::test_sparse_realdata
+
+  # Different models between oneAPI Data Analytics Library (oneDAL) and stock sklearn with callble kernel
```
Comment (OnlyDeniko) [inline deselected_tests.yaml:35]:
```suggestion
  # Different models between oneAPI Data Analytics Library (oneDAL) and stock sklearn with callable kernel
```

**Comment by PetrovKP** [comment]
/intelci: run
set infra_branch=sklearnex-tests

**Comment by PetrovKP** [comment]
/intelci: run
set infra_branch=sklearnex-tests

**Comment by PetrovKP** [comment]
/intelci: run
set infra_branch=sklearnex-tests

**Comment by PetrovKP** [comment]
/intelci: run
set infra_branch=sklearnex-tests

**Comment by PetrovKP** [comment]
/intelci: run
set infra_branch=sklearnex-tests

**Comment by PetrovKP** [comment]
/intelci: run
set infra_branch=sklearnex-tests

**Comment by PetrovKP** [comment]
/intelci: run
set infra_branch=sklearnex-tests

**Comment by PetrovKP** [comment]
/intelci: run
set infra_branch=sklearnex-tests

**Comment by PetrovKP** [comment]
/intelci: run
set infra_branch=sklearnex-tests

**Comment by PetrovKP** [comment]
/intelci: run
set infra_branch=sklearnex-tests
set ext_build=ebb7c5d4-b3de-f1b1-a0fd-8cdcd4b723a1

**Comment by PetrovKP** [comment]
/intelci: rustart
set infra_branch=sklearnex-tests
set ext_build=ebb7c5d4-b3de-f1b1-a0fd-8cdcd4b723a1

**Comment by PetrovKP** [comment]
/intelci: restart
set infra_branch=sklearnex-tests
set ext_build=ebb7c5d4-b3de-f1b1-a0fd-8cdcd4b723a1

**Comment by PetrovKP** [comment]
/intelci: run
set infra_branch=sklearnex-tests
set ext_build=ebb7c5d4-b3de-f1b1-a0fd-8cdcd4b723a1

**Review by Pahandrovich — APPROVED** [review]
:+1:

**Comment by PetrovKP** [comment]
/intelci: run
set infra_branch=sklearnex-tests
set ext_build=ebb7c5d4-b3de-f1b1-a0fd-8cdcd4b723a1

**Comment by PetrovKP** [comment]
/intelci: run
set infra_branch=sklearnex-tests
set ext_build=ebb7c5d4-b3de-f1b1-a0fd-8cdcd4b723a1

**Comment by owerbat** [comment]
Can't push the approve button. But it looks good for me (in case you don't forget to support SCR tables) 👍

**Comment by PetrovKP** [comment]
/intelci: run
set infra_branch=sklearnex-tests
set ext_build=ebb7c5d4-b3de-f1b1-a0fd-8cdcd4b723a1

**Comment by PetrovKP** [comment]
/intelci: run
set infra_branch=sklearnex-tests

**Pull request merged**